src/aspire/utils/misc.py

```diff
@@ -303,29 +303,19 @@ def fuzzy_mask(L, r0, risetime, origin=None):
 
 def all_pairs(n):
     """
-    All pairs indexing (i,j) for i<j.
+    All pairs indexing (i,j) for i<j and a pairs-to-linear index mapping.
 
     :param n: The number of items to be indexed.
-    :return: All n-choose-2 pairs (i,j), i<j.
+    :returns:
+        - n x 2 array of pairs (i, j), i<j.
+        - dictionary mapping pairs to linear indices.
     """
     pairs = np.column_stack(np.triu_indices(n, 1))
+    pairs_to_linear_map = np.empty((n, n), dtype=np.uint16)
```

Comment:
I haven't checked, but this might be even faster to generate.

`np.put(pairs_to_linear_map, pairs, np.arange(len(pairs)))`